utils/common/src/main/java/org/apache/brooklyn/util/collections/CollectionFunctionals.java

```diff
@@ -119,6 +115,35 @@ public T apply(Iterable<? extends T> input) {
         @Override public String toString() { return "firstElementFunction"; }
     }
 
+    public static class AllEqualsFunction implements Function<Iterable<?>, Comparable<?>> {
+        private final Comparable<?> value;
+
+        public AllEqualsFunction(Comparable<?> value) {
+            if (Objects.isNull(value)) {
+                throw new IllegalArgumentException("The value to compare all to must be defined");
+            }
+            this.value = value;
+        }
+
+        @Override
+        public Boolean apply(@Nullable Iterable<?> input) {
+            if (Objects.isNull(input) || Iterables.isEmpty(input)) return null;
```

Comment:
I think it should return `false` if the input is empty or null unless the comparable is an empty iterable or a null value.

Not sure how functional is my logic about that

Reply:
Moreover, if there is a DSL watching this `$brooklyn:attributeWhenReady("service.process.isRunning")`, then it will wait until there is a value to read.

Reply:
This should address your doubt @jcabrerizo

Reply:
Agree, but for me, it looks like we need to have flag for do not evaluate unless is not empty or not null. 
Yes, the use of `brooklyn:attributeWhenReady` will help, but if only the function used, the transformer will return null if the source sensor is not returning a false value.

I'm happy to approve but please, add some comments explaining why it will return `null` comparing with an empty collection

Reply:
Documented in 568af812d307c52c6cb7f1abbb6e363526ddd3ec